python/pyspark/sql/dataframe.py

```diff
@@ -968,7 +968,7 @@ def hint(
         if not isinstance(name, str):
             raise TypeError("name should be provided as str, got {0}".format(type(name)))
 
-        allowed_types = (str, list, float, int)
+        allowed_types = (str, Column, int)
```

Comment:
Maybe we can add `Column` into this allowed list, and leave things as are because other custom types would be broken without this. BTW, does Scala support such hint too? e.g, `df.hint("REPARTITION", "a")`. This method calls Scala side directly.